client/src/lib/attempt-result.js

```diff
@@ -483,3 +489,15 @@ export function meetsCutoff(attemptResults, cutoff) {
     .slice(0, numberOfAttempts)
     .some((attempt) => attempt > 0 && attempt < attemptResult);
 }
+
+/**
+ * Checks if the given attempt results contain a DNS followed by a valid result.
+ */
+export function checkForDnsFollowedByValidResult(attemptResults) {
```

Comment:
Currently this is just an implementation detail of the warning function, so I would keep it private. If the idea is to export it for testing, usually it's better to test the caller (here, the warning function).

```suggestion
function checkForDnsFollowedByValidResult(attemptResults) {
```